cmd/kubeadm/app/apis/kubeadm/v1beta2/public_key_algorithm.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/pkg/errors"
+	kubeadm "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+)
+
+// PublicKeyAlgorithm is a wrapper around kubeadm.PublicKeyAlgorithm which supports
+// correct marshaling to YAML and JSON. In particular, it marshals into strings.
```

Comment:
instead of using `int` in the internal config (also not use the go enumeration) we might as well use "string" in the internal configuration and this can avoid having this file.

it would still need to convert from the go enumeration to our string enumeration.

Reply:
Indeed, we don't need this file. Let's use strings and string consts here.